The job **`dist-arm-linux`** failed! Check out the build log: [(web)](https://github.com/rust-lang-ci/rust/actions/runs/9924797882/job/27416516093) [(plain)](https://github.com/rust-lang-ci/rust/commit/f24522ae7b6427c2ca94da8e493e0126025acbda/checks/27416516093/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
[RUSTC-TIMING] syn test:false 3.561
   Compiling semver v1.0.23
error[E0463]: can't find crate for `core`
  |
  = note: the `arm-unknown-linux-gnueabi` target may not be installed
  = help: consider downloading the target with `rustup target add arm-unknown-linux-gnueabi`
  = help: consider building the standard library from source with `cargo build -Zbuild-std`
   Compiling equivalent v1.0.1
For more information about this error, try `rustc --explain E0463`.
[RUSTC-TIMING] cfg_if test:false 0.022
error: could not compile `cfg-if` (lib) due to 1 previous error
error: could not compile `cfg-if` (lib) due to 1 previous error
warning: build failed, waiting for other jobs to finish...
[RUSTC-TIMING] once_cell test:false 0.026
error: could not compile `once_cell` (lib) due to 1 previous error
error[E0463]: can't find crate for `std`
  |
  = note: the `arm-unknown-linux-gnueabi` target may not be installed
  = help: consider downloading the target with `rustup target add arm-unknown-linux-gnueabi`
  = help: consider building the standard library from source with `cargo build -Zbuild-std`
[RUSTC-TIMING] bitflags test:false 0.031
error: could not compile `bitflags` (lib) due to 1 previous error
[RUSTC-TIMING] zerocopy test:false 0.036
error: could not compile `zerocopy` (lib) due to 1 previous error
```

</details>

--- a/src/bootstrap/src/core/build_steps/compile.rs
+++ b/src/bootstrap/src/core/build_steps/compile.rs
@@ -1837,7 +1837,7 @@ impl Step for Assemble {
             // `wasm32-wasip2` target of Rust.
             let wasm_component_ld_exe =
                 builder.ensure(crate::core::build_steps::tool::WasmComponentLd {
-                    compiler: build_compiler.with_stage(0),
+                    compiler: build_compiler,
                     target: target_compiler.host,
                 });
             builder.copy_link(

--- a/src/bootstrap/src/core/build_steps/tool.rs
+++ b/src/bootstrap/src/core/build_steps/tool.rs
@@ -337,7 +337,7 @@ bootstrap_tool!(
     RustdocGUITest, "src/tools/rustdoc-gui-test", "rustdoc-gui-test", is_unstable_tool = true, allow_features = "test";
     CoverageDump, "src/tools/coverage-dump", "coverage-dump";
     RustcPerfWrapper, "src/tools/rustc-perf-wrapper", "rustc-perf-wrapper";
-    WasmComponentLd, "src/tools/wasm-component-ld", "wasm-component-ld";
+    WasmComponentLd, "src/tools/wasm-component-ld", "wasm-component-ld", is_unstable_tool = true, allow_features = "min_specialization";
 );
 
 #[derive(Debug, Clone, Hash, PartialEq, Eq)]

